core/include/arch/thread_arch.h

```diff
@@ -1,96 +0,0 @@
-/*
- * Copyright (C) 2014 Freie Universität Berlin
- *
- * This file is subject to the terms and conditions of the GNU Lesser
- * General Public License v2.1. See the file LICENSE in the top level
- * directory for more details.
- */
-
-/**
- * @ingroup     core_arch
- * @{
- *
- * @file
- * @brief       Architecture dependent kernel interface for handling and managing threads
- *
- * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
- */
-
-#ifndef ARCH_THREAD_ARCH_H
-#define ARCH_THREAD_ARCH_H
-
-#include "kernel_defines.h"
-
-#ifdef __cplusplus
- extern "C" {
-#endif
-
-/**
- * @name Define the mapping between the architecture independent interfaces
- *       and the kernel internal interfaces
- *
- * This mapping is done for compatibility of existing platforms,
- * new platforms should always use the *_arch_* interfaces.
- * @{
- */
-#ifdef COREIF_NG
-#define thread_stack_init               thread_arch_stack_init
-#define thread_print_stack              thread_arch_stack_print
-#define cpu_switch_context_exit         thread_arch_start_threading
```

Comment:
While at it: can we please rename this function according to our naming schemes (module name as prefix). Now I am actually unsure if this is part of the scheduler (also its defined in `sched.h`), or if this functions is rather logically part of the `thread` module in the same context as the other functions in this block?!

Reply:
WHILE AT IT? ;)

I'd rather keep the PR as small as possible.

Reply:
ok